benchmark/spmv/batch_spmv.cpp

```diff
@@ -0,0 +1,484 @@
+                        std::string b_str;
+                        if (FLAGS_batch_scaling == "implicit") {
+                            b_str = "b_scaled.mtx";
+                        } else {
+                            b_str = "b.mtx";
+                        }
+                        std::string fname =
+                            std::string(test_case["problem"].GetString()) +
+                            "/" + std::to_string(i) + "/" + b_str;
+                        std::ifstream b_fd(fname);
+                        bdata[i] = gko::read_raw<etype>(b_fd);
+                    }
+                    auto temp_b = batch_vec<etype>::create(exec);
+                    temp_b->read(bdata);
+                    b = batch_vec<etype>::create(exec, ndup, temp_b.get());
+                } else {
+                    b = create_batch_matrix<etype>(
+                        exec,
+                        gko::batch_dim<2>(nbatch * multiplier,
+                                          gko::dim<2>{data[0].size[0], nrhs}),
+                        engine);
+                }
+            }
+            auto x = create_batch_matrix<etype>(
+                exec,
+                gko::batch_dim<2>(nbatch * multiplier,
+                                  gko::dim<2>{data[0].size[0], nrhs}),
+                engine);
+
+            // Compute the result from ginkgo::batch_csr as the correct answer
+            auto answer = batch_vec<etype>::create(exec);
+            if (FLAGS_detailed) {
+                std::shared_ptr<gko::BatchLinOp> system_matrix;
+                if (FLAGS_using_suite_sparse) {
+                    system_matrix = share(formats::batch_matrix_factory.at(
+                        "batch_csr")(exec, ndup, data[0]));
+                } else {
+                    system_matrix = share(formats::batch_matrix_factory2.at(
+                        "batch_csr")(exec, ndup, data));
+                }
+                auto sys_size = system_matrix->get_size();
+                auto x_size = x->get_size();
+                auto b_size = b->get_size();
+                answer->copy_from(lend(x));
+                exec->synchronize();
+                system_matrix->apply(lend(b), lend(answer));
+                exec->synchronize();
+            }
+            for (const auto &format_name : formats) {
+                if (FLAGS_using_suite_sparse) {
+                    apply_spmv(format_name.c_str(), exec, data[0], lend(b),
+                               lend(x), lend(answer), test_case, allocator);
+                } else {
+                    apply_spmv(format_name.c_str(), exec, data, lend(b),
+                               lend(x), lend(answer), test_case, allocator);
+                }
+                std::clog << "Current state:" << std::endl
+                          << test_cases << std::endl;
+                if (spmv_case[format_name.c_str()]["completed"].GetBool()) {
+                    auto performance =
```

Comment:
Is there something missing here? It looks like `performance` will be simply be deleted.

Reply:
Yes, you are right. This was for showing the best spmv format, which is not relevant for batch matrices, atleast not right now.